Dockerfile

```diff
@@ -0,0 +1,26 @@
+# Build ditto
+# docker build . -t ditto
+
+# Run ditto
+# docker run --rm -ti ditto --help
+
+# https://hub.docker.com/_/python
+FROM python:3.6-slim-bullseye
+
+# Install latest version of ditto
+RUN apt-get update; apt-get install -y git
+RUN git clone --depth 1 https://github.com/NREL/ditto.git
```

Comment:
I'm pretty sure you can also just copy files from the current directory into the working directory, and don't need to install `git` and clone the repo. Presumably any user will have cloned the repo already to build the docker image.

Reply:
This is true, though cloning the repo makes the Dockerfile more portable.

Reply:
Sure. I think that’s useful for when we want to just share the Dockerfile, but if someone wants to develop and test locally with docker this wouldn’t work, right? 

I don’t feel strongly about this and I have little docker experience. I’m happy to defer to you on what you think is best practice. Maybe do a quick search for other docker files from popular python projects and see what they do?

If you do decide to leave it this way, can you uninstall `git` after cloning?

Reply:
@kdheepak Updated in 61a8c8f

```bash
❯ time docker build . -t ditto --no-cache=true
[+] Building 122.3s (11/11) FINISHED
 => [internal] load build definition from Dockerfile                                                                                                                                                   0.1s
 => => transferring dockerfile: 37B                                                                                                                                                                    0.0s
 => [internal] load .dockerignore                                                                                                                                                                      0.0s
 => => transferring context: 2B                                                                                                                                                                        0.0s
 => [internal] load metadata for docker.io/library/python:3.6-slim                                                                                                                                     1.2s
 => [internal] load build context                                                                                                                                                                      0.6s
 => => transferring context: 483.09kB                                                                                                                                                                  0.6s
 => CACHED [1/6] FROM docker.io/library/python:3.6-slim@sha256:a7989f57b5078eae628f484fc39e249c88fd3404427e100aabcc5c6bc07cc1e7                                                                        0.0s
 => [2/6] COPY ./ /ditto                                                                                                                                                                               2.9s
 => [3/6] WORKDIR /ditto                                                                                                                                                                               0.1s
 => [4/6] RUN python -m pip install --upgrade pip && pip install -e .[all] && pip install pytest                                                                                                      51.3s
 => [5/6] RUN pytest -sv                                                                                                                                                                              48.8s
 => [6/6] RUN ditto --help                                                                                                                                                                             1.9s
 => exporting to image                                                                                                                                                                                15.3s
 => => exporting layers                                                                                                                                                                               15.3s
 => => writing image sha256:01aa1ffe4269ce3b5210c751cda5bc8808bf273f42b2842b39b4764d49a5c9af                                                                                                           0.0s
 => => naming to docker.io/library/ditto                                                                                                                                                               0.0s

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
docker build . -t ditto --no-cache=true  0.99s user 1.30s system 1% cpu 2:04.50 total
```